firebase-firestore/src/main/java/com/google/firebase/firestore/util/Preconditions.java

```diff
@@ -14,11 +14,87 @@
 
 package com.google.firebase.firestore.util;
 
-import static com.google.firebase.firestore.util.Strings.lenientFormat;
-
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
+/**
+ * This is an adaptation of
+ * https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Preconditions.java.
```

Comment:
Same comment regarding the placement of this.

Reply:
Done.